source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -233,22 +234,14 @@ void AuthenticatorImpl::verifyKey() {
 }
 
 bool AuthenticatorImpl::okToBypass() const {
-  if (config_->getProtoConfig().allow_missing_or_failed()) {
-    return true;
-  }
-
   // TODO(qiwzhang): use requirement field
   return false;
 }
 
 void AuthenticatorImpl::doneWithStatus(const Status& status) {
   ENVOY_LOG(debug, "Jwt authentication completed with: {}",
             ::google::jwt_verify::getStatusString(status));
-  if (status != Status::Ok && config_->getProtoConfig().allow_missing_or_failed()) {
-    callback_->onComplete(Status::Ok);
-  } else {
-    callback_->onComplete(status);
-  }
+  callback_->onComplete(status);
```

Comment:
For use by other filters, is it possible to have an onComplete callback that passes a const reference/pointer to the token as well as the status? In my uses case (OpenID Connect) I need to check some additional JWT claims to validate against XSRF attacks. These claims are potentially use case specific so pushing them into the Authenticator would be wrong.

Reply:
Clearly, you were thinking about linking jwt_authn code into your filter. 

I will prefer to explore if you can use jwt_authn code as a filter.  Your filter can use its verification result.   Istio_authn originally linked jwt_authn code into its filter,  it makes the filter very complicated since it needs to deal with aync callbacks.  We abandoned that idea.  Now it will not touch jwt_authn filter, just config it to behave the way istio_authn wanted and uses its result.

Now the result is passed by HTTP headers.  We can change it to use requestinfo.dynamidMetadata to pass the result. https://github.com/envoyproxy/envoy/blob/master/include/envoy/request_info/request_info.h#L282

Reply:
In (near) future we will make the verified JWT into metadata, that will allow JWT works with RBAC filter, see #3638 too. I think that will address your use case? @nickrmc83

Reply:
@qiwzhang @lizan see my earlier comment about why I can't really re-use the code as a stand-alone filter and why I would like to directly invoke the Authenticator verification code. I suggest pushing forward this PR and discussing my usage in a PR I raise as I think my needs will become clearer.